nncf/quantization/algorithms/smooth_quant/algorithm.py

```diff
@@ -161,7 +161,8 @@ def apply(
                 weight_update_command = self._backend_entity.weight_update_command(node_to_smooth, scaled_weight.data)
                 transformation_layout.register(weight_update_command)
 
-            activations_shape = graph.get_output_edges(source_node)[source_output_port_id].tensor_shape
```

Comment:
This behaviour that utilizes the `get_output_edges` order may be the reason for the other issues in different places in NNCF. As the common approach, only identification by the needed `input-`/`output_port_id` field should be used.
@alexsu52